common.py

```diff
@@ -855,41 +858,140 @@ def read_training_features(model_file):
     C, RHO_edges, PT_edges = np.histogram2d(rho[cuts], pt[cuts], bins=49,weights=weight[cuts])
+
+    # Initialize a 2D map for the variable
     w, h = 50, 50
-    VAR_map      = [[0 for x in range(w)] for y in range(h)]
+    VAR_map = [[0 for x in range(w)] for y in range(h)]
+
+    # Get the variable for the data that passed the cuts
     VAR = var[cuts]
+
+    # Loop over the bins in rho and pt
     for i in range(len(RHO_edges)-1):
-       for j in range(len(PT_edges)-1):
-          CUT = (rho[cuts]>RHO_edges[i]) & (rho[cuts]<RHO_edges[i+1]) & (pt[cuts]>PT_edges[j]) & (pt[cuts]<PT_edges[j+1])
-          if len(VAR[CUT])==0: continue
-          if len(VAR[CUT])>0:
-             VAR_map[i][j]=np.percentile(VAR[CUT],100-percent) # percent is calculated based on the bdt working point
+        for j in range(len(PT_edges)-1):
+            # Apply cuts to select data in the current rho and pt bin
+            CUT = (rho[cuts]>RHO_edges[i]) & (rho[cuts]<RHO_edges[i+1]) & (pt[cuts]>PT_edges[j]) & (pt[cuts]<PT_edges[j+1])
+
+            # If there is no data in this bin, skip it
+            if len(VAR[CUT])==0: continue
 
+            # If there is data in this bin, calculate the percentile of the variable
+            if len(VAR[CUT])>0:
+                VAR_map[i][j]=np.percentile(VAR[CUT],100-percent) # percent is calculated based on the bdt working point
+
+    # Smooth the variable map using a Gaussian filter
     VAR_map_smooth = gaussian_filter(VAR_map,1)
+
+    # Return the smoothed variable map, along with the rho and pt edges
     return VAR_map_smooth, RHO_edges, PT_edges
 
-def ddt(mt, pt, rho, var, weight, percent):
+# Class that converts numpy arrays into list so they can be easily stored in json files
+class NumpyArrayEncoder(json.JSONEncoder):
+    """Custom encoder for numpy data types."""
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return super(NumpyArrayEncoder, self).default(obj)
+
+def create_DDT_map_dict(mt, pt, rho, var, weight, percents, cut_vals, ddt_name):
+    '''
+    This function creates the dictionary of DDT 2D maps for a range of
+        cut_vals at corresponding bkg efficiencies given as percents
+    The DDT 2D map at each cut_val contains: var_map_smooth, RHO_edges, and PT_edges, 
+        The inputs to the function are (mt, pt, rho, var, weight, percents, cut_vals, ddt_name).
+    In essense the DDT is a 2D function of rho and pt that
+        is calculated using the background data for the variable
+        that you want to decorrelate
+    '''
+
+    # Apply the rho-ddt window cuts to the data
+    cuts = rhoddt_windowcuts(mt, pt, rho)
+
+    # Initialize the dictionary
+    var_dict = {}
+
+    # Loop over the values in cut_vals and percents
+    for cut_val, percent in zip(cut_vals, percents):
+
+        print("Creating DDT 2D map for a cut value of ", cut_val, " and an efficiency (percent) of ", percent)
```

Comment:
should be able to disable this printout too

Reply:
Added